config/settings.go

```diff
@@ -0,0 +1,114 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+func ShouldUploadBuildLogs() bool {
+
+	if v, err := GetConfigValue("user", "upload_build_logs"); err == nil {
+		if v == "true" {
+			return true
+		}
+	}
+
+	return false
+}
+
+func YourBaseProfile() string {
+	profile, exists := os.LookupEnv("YOURBASE_PROFILE")
+
+	if exists {
+		return profile
+	}
+
+	profile, exists = os.LookupEnv("YB_PROFILE")
+
+	if exists {
+		return profile
+	}
+
+	return ""
+}
+
+func apiBaseUrl() string {
+	if url, exists := os.LookupEnv("YOURBASE_API_URL"); exists {
+		return url
+	}
+
+	if url, err := GetConfigValue("yourbase", "api_url"); err == nil {
+		if url != "" {
+			return url
+		}
+	}
+
+	profile := YourBaseProfile()
+
+	switch profile {
+	case "staging":
+		return "https://api.staging.yourbase.io"
+	case "development":
+		return "http://localhost:5000"
+	default:
```

Comment:
Maybe it's more helpful / user friendly to be more strict. I would do:

case "production":
case " ":
  return prod-stuff, nil

default:
   return ''
", error

That sadly means changing all call paths of apiBaseURL to take an error value too but it could save some frustration for someone trying to understand why the "dev" profile (or another invalid value) doesn't work.